console/console-init/ui/src/modules/iot-device-detail/components/CredentialsView/CredentialsView.tsx

```diff
@@ -92,89 +92,80 @@ export const Credential: React.FC<ICredentialProps> = ({
-                id="credentials-view-status-title"
-              >
-                Status
-              </Title>
-            </GridItem>
-            <GridItem span={3}>
-              <Title
-                headingLevel="h1"
-                size="md"
-                id="credentials-view-enable-title"
-              >
-                Enable
-              </Title>
-            </GridItem>
-            <GridItem span={9}>
-              {enableActions ? (
-                <Switch
-                  id={"credentials-view-status-switch-button"}
-                  label={"On"}
-                  labelOff={"Off"}
-                  isChecked={enabled}
-                  onChange={onChange}
-                />
-              ) : enabled ? (
-                <>
-                  <CheckCircleIcon color="green" /> &nbsp; On
-                </>
-              ) : (
-                <StatusLabelWithIcon
-                  id="credential-view-status-icon"
-                  key={`credential-view-status-${authId}`}
-                  isEnabled={enabled || false}
-                  disabledTitle={"Off"}
-                  enabledTitle={"On"}
-                  disabledIconColor="var(--pf-global--palette--black-300)"
-                />
-              )}
-            </GridItem>
-          </>
-        )}
-      </Grid>
-    </>
+    <Grid>
+      <GridItem span={3}>
+        <Title headingLevel="h1" size="md" id="credentials-view-auth-id-title">
+          {getLabelByKey("auth-id")}
+        </Title>
+      </GridItem>
+      <GridItem span={9} className={css(styles.row_margin)}>
+        {authId}
+      </GridItem>
+      <GridItem span={3}>
+        <Title headingLevel="h1" size="md" id="credentials-view-type-title">
+          {getLabelByKey("type")}
+        </Title>
+      </GridItem>
+      <GridItem span={9} className={css(styles.row_margin)}>
+        {getLabelByKey(type || "")}
+      </GridItem>
+      <SecretsViewContainer
+        secrets={secrets}
+        id={"credetials-view-secrets-" + authId}
+        heading={"Secrets"}
+        authId={authId}
+        credentialType={type}
+        enableActions={enableActions}
+      />
+      <ExtensionsView
+        id={"credetials-view-extensions-" + authId}
+        ext={ext}
+        heading={"Ext"}
+      />
+      {hasOwnProperty(credential, "enabled") && (
+        <>
+          <GridItem span={12} className={css(styles.status_section_margin)}>
+            <Title
+              headingLevel="h6"
+              size="xl"
+              id="credentials-view-status-title"
+            >
+              Status
+            </Title>
+          </GridItem>
+          <GridItem span={3}>
+            <Title
+              headingLevel="h1"
+              size="md"
+              id="credentials-view-enable-title"
+            >
+              Enable
+            </Title>
+          </GridItem>
+          <GridItem span={9}>
+            {enableActions ? (
+              <Switch
```

Comment:
Please use the generic SwitchWithToggle if possible.

Reply:
SwitchWithToggle is not possible here.

Reply:
Any specific reason?